### ISSUE TITLE
Ban bogus "existential types"

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#7084]
It is possible to get an unbound type variable in a variable type using a generic alias, for example:
```python
from typing import TypeVar, Dict
  
T = TypeVar('T')
Alias = Dict[str, T]

x: Alias[T]
reveal_type(x)  # Revealed type is 'builtins.dict[builtins.str, T?]'
```
This doesn't makes sense currently and should give an error.

See also https://github.com/python/mypy/issues/6520 (this looks very similar but is actually a different issue specific to type aliases).
